File: solutions-cpp/src/main/cpp/dsalgo/string/LC_0387_first_unique_char.cpp
#include <unordered_map>
#include <string>

int firstUniqChar(std::string s) {
    std::unordered_map<char, int> charCount;
    // Count the frequency of each character
    for (char c : s) {
        charCount[c]++;
    }
    // Find the first unique character
    for (int i = 0; i < s.size(); i++) {
        if (charCount[s[i]] == 1) {
            return i;
        }
    }
    return -1;
}

File: solutions-go/src/main/strings/LC_0387_first_unique_char.go
package strings;

func FirstUniqChar(s string) int {
    charCount := make(map[rune]int)

    // Count the occurrences of each character
    for _, char := range s {
        charCount[char]++
    }

    // Find the index of the first unique character
    for i, char := range s {
        if charCount[char] == 1 {
            return i
        }
    }

    return -1
}
